src/main/java/com/browserstack/automate/ci/common/uploader/AppUploader.java

```diff
@@ -12,18 +13,32 @@
 
     String appPath;
     BrowserStackCredentials credentials;
+    private final transient PrintStream logger;
+    private final String customProxy;
 
-    public AppUploader(String appPath, BrowserStackCredentials credentials) {
+    public AppUploader(String appPath, BrowserStackCredentials credentials, final String customProxy, final PrintStream logger) {
         this.appPath = appPath;
         this.credentials = credentials;
+        this.logger = logger;
+        this.customProxy = customProxy;
     }
 
     public String uploadFile()
             throws AppAutomateException, FileNotFoundException, InvalidFileExtensionException {
         AppAutomateClient appAutomateClient =
                 new AppAutomateClient(credentials.getUsername(), credentials.getDecryptedAccesskey());
-        if (JenkinsProxySettings.hasProxy()) {
-            appAutomateClient.setProxy(JenkinsProxySettings.getHost(), JenkinsProxySettings.getPort(), JenkinsProxySettings.getUsername(), JenkinsProxySettings.getPassword());
+
+        JenkinsProxySettings proxy;
+        if (customProxy != null) {
+            proxy = new JenkinsProxySettings(customProxy, logger);
+        } else {
+            proxy = new JenkinsProxySettings(logger);
+        }
+
+        if (proxy.hasProxy()) {
+            System.out.println("App upload setting proxy for app automate client...");
```

Comment:
Logs

Reply:
Yep. Will remove the logs in one commit.

Reply:
Removed the logs in [
8291572](https://github.com/jenkinsci/browserstack-integration-plugin/pull/30/commits/82915726cafc29f8bf88dce8fd8bacc6f450cbfc)

Reply:
Resolving